edge/stack/stack.go

```diff
@@ -260,6 +262,57 @@ func (manager *StackManager) nextPendingStack() *edgeStack {
 			return stack
 		}
 	}
+
+	for _, stack := range manager.stacks {
+		if stack.Status == StatusRetry {
+			stack.Status = StatusPending
+		}
+	}
+
+	return nil
+}
+
+func (manager *StackManager) pullImages(ctx context.Context, stack *edgeStack, stackName, stackFileLocation string) error {
+	manager.mu.Lock()
+	defer manager.mu.Unlock()
+
+	log.Debug().Int("stack_identifier", int(stack.ID)).Msg("stack pulling images")
+
+	if stack.PrePullImage || stack.RePullImage {
+		stack.Retries += 1
+		if stack.Retries <= RetryInterval || stack.Retries%RetryInterval == 0 {
```

Comment:
It seems that the function will retry more than 3 times for the worst case?

Reply:
Roughly, it will try a few days and once per hour.